shared/pkg/families/manager.go

```diff
@@ -77,17 +78,35 @@ func New(logger *log.Entry, config *Config) *Manager {
 
 type RunErrors map[types.FamilyType]error
 
-func (m *Manager) Run() (*results.Results, RunErrors) {
-	familyErrors := make(map[types.FamilyType]error)
+type familyResult struct {
+	result interfaces.IsResults
+	err    error
+}
 
+func (m *Manager) Run(ctx context.Context) (*results.Results, RunErrors) {
+	familyErrors := make(RunErrors)
 	familyResults := results.New()
+	result := make(chan familyResult, len(m.families))
```

Comment:
I'm not sure but from a memory management perspective it might be better to move this channel into the loop, so that we can clean it up when we're done for example:
```
for _, family := range m.families {
    result := make(chan familyResult)
    go func() {
        ret, err := family.Run(familyResults)
        result <- familyResult{
           ret,
           err,
        }
    }()
    
    select {
       case <-ctx.Done():
          // drain and close result channel
          go func() {
              <-result
              close(result)
          }()
          // handle aborted error
        case <-result:
           // handle result
           close(result)
    }
}
```

Reply:
So the reason why `result` channel is created outside the for loop is because I think that certain families can be grouped together and run them parallel.
But I will add suggested solution and will deal with making families run parallel later.

Reply:
Fixed.